packages/react/src/components/TextField/TextField.tsx

```diff
@@ -16,29 +16,108 @@
  * under the License.
  */
 
-import {InputLabel} from '@mui/material';
+import InputAdornment from '@mui/material/InputAdornment';
 import MuiTextField, {TextFieldProps as MuiTextFieldProps} from '@mui/material/TextField';
+import {DoubleCircleIcon, VisibilityIcon, VisibilityOffIcon} from '@oxygen-ui/react-icons';
 import clsx from 'clsx';
-import {FC, ReactElement} from 'react';
+import {FC, MouseEvent, ReactElement, ReactNode, useState} from 'react';
 import {WithWrapperProps} from '../../models';
 import {composeComponentDisplayName} from '../../utils';
+import IconButton from '../IconButton';
+import InputLabel from '../InputLabel';
+import List from '../List';
+import ListItem from '../ListItem';
+import ListItemIcon from '../ListItemIcon';
+import ListItemText from '../ListItemText';
+import Tooltip from '../Tooltip';
 import './text-field.scss';
 
-export type TextFieldProps = MuiTextFieldProps;
+export type TextFieldProps = {
+  /**
+   * Criteria for user input.
+   */
+  criteria?: string[];
+} & MuiTextFieldProps;
 
 const COMPONENT_NAME: string = 'TextField';
 
 const TextField: FC<TextFieldProps> & WithWrapperProps = (props: TextFieldProps): ReactElement => {
-  const {className, label, InputLabelProps, id, ...rest} = props;
+  const {className, criteria, InputLabelProps, id, label, type, ...rest} = props;
 
   const classes: string = clsx('oxygen-text-field', className);
 
+  const [open, setOpen] = useState<boolean>(false);
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClick = (): void => {
+    if (type === 'password' && criteria?.length > 0) {
+      setOpen(true);
+    }
+  };
+
+  const handleClose = (): void => {
+    if (open) {
+      setOpen(false);
+    }
+  };
+
+  const handleClickShowPassword = (): void => setShowPassword((show: boolean) => !show);
+
+  const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>): void => {
+    event.preventDefault();
+  };
+
+  const tooltipContent = (): ReactNode => (
+    <List>
+      {criteria?.map((criterion: string) => (
+        <ListItem disablePadding>
+          <ListItemIcon>
+            <DoubleCircleIcon />
+          </ListItemIcon>
+          <ListItemText primary={criterion} />
+        </ListItem>
+      ))}
+    </List>
+  );
+
   return (
     <div className={classes}>
-      <InputLabel htmlFor="id" shrink className="oxygen-text-field__label" {...InputLabelProps}>
+      <InputLabel htmlFor={id} aria-describedby={id} {...InputLabelProps}>
         {label}
       </InputLabel>
-      <MuiTextField className="oxygen-text-field__input" id={id} {...rest} />
+      <Tooltip
```

Comment:
Will the tooltip appear in the DOM tree even if no criteria are passed in?

Reply:
Fixed by isolating the components